exercises/practice/two-bucket/source/two_bucket.d

```diff
@@ -0,0 +1,83 @@
+module two_bucket;
+
+import std.typecons;
+
+alias TwoBucketInput = Tuple!(
+        int, "size1",
+        int, "size2",
+        int, "goal",
+        string, "startBucket",
+);
+
+alias TwoBucketResult = Tuple!(
+        int, "moves",
+        string, "goalBucket",
+        int, "otherAmount",
+);
+
+pure TwoBucketResult measure(TwoBucketInput input)
+{
+    // Implement this function
+}
+
+unittest
+{
+    import std.exception : assertThrown;
+
+    immutable int allTestsEnabled = 0;
+
+    // Measure using bucket one of size 3 and bucket two of size 5 - start with bucket one
+    {
+        auto result = measure(TwoBucketInput(3, 5, 1, "one"));
+        assert(result == TwoBucketResult(4, "one", 5));
+    }
+
+    //static if (allTestsEnabled)
```

Comment:
```suggestion
    static if (allTestsEnabled)
```

All tests will be run for the student by default with this commented out.

Reply:
woops